libs/core/src/lib/carousel/carousel.component.ts

```diff
@@ -321,7 +325,17 @@ export class CarouselComponent implements OnInit, AfterContentInit, AfterViewIni
         this._adjustActiveItemPosition(SlideDirection.PREVIOUS);
         this._preventDefaultBtnFocus();
         this._carouselService.pickPrevious(this.dir);
-        this._notifySlideChange(SlideDirection.PREVIOUS);
+
+        /** Handle looped carousel, first click on prev button. */
+        if (this.loop && this._prevBtnClickedAtStart) {
+            this._carouselService.goToItem(this.slides.toArray()[Math.ceil(this.slides.length / 2) - 1], true, this.dir);
```

Comment:
Looks how many times you are creating new slides array 😛
Get the value once and use it.

Reply:
fixed.